### DISCUSSION TITLE
Minor change to native `Model` class to fix some IDE issues

### DISCUSSION THREAD

**Discussion by BrekiTomasson** [laravel/framework#43858] [Ideas]
One of the biggest reason people reach for tools like the Laravel IDE Helper is because of how most IDE:s don't natively understand that Models can use Builder methods, meaning that as soon as you type that initial `::where('foo', 'bar')` you get that dreaded yellow line of "Method 'where' not found in \App\Models\Auth\User". For the longest time, I've been using a hacky workaround by simply creating my own `\App\Models\Model` class that extends `Illuminate\Database\Eloquent\Model` and then adding this DocBlock to the entire class:

```php
/**
 * @mixin \Illuminate\Database\Eloquent\Builder
 */
```

That way, whenever I create a new model, I can extend my own `Model` class instead of Laravel's `Model` class, and my IDE understands that `Builder` methods should be available as well. Now, as you can imagine, this fails as soon as I need to use a third party package's models, as they obviously extend Laravel's model class instead of my own, so that's got me thinking...

... why isn't this simple PHP DocBlock part of Laravel's code already? I can't imagine it would break anything, and it would be a much more "native" way of solving the problem of the IDE finding your magic methods than installing a third party package that creates a separate file to trick the IDE, right?

**Comment by Sophist-UK** [comment]
Why not submit a PR to add the DocBlock?

**Reply by BrekiTomasson** [reply]
I will, I was mainly just curious to see if smarter brains than mine had thoughts about this before doing so, in case I was missing something obvious.

**Comment by allanlaal** [comment]
a more macro level issue is using static typing in Laravel core: https://github.com/laravel/framework/discussions/36595

**Comment by allanlaal** [comment]
I solved mine with a custom base Model class aswell:

	use Illuminate\Contracts\Database\Eloquent\Builder as BuilderContract;
	use Illuminate\Database\Eloquent\Model as LaravelModel;

	abstract class Model extends LaravelModel implements BuilderContract {}
